hw18.1

Mykhaylo12 opened the pull request.

Mykhaylo12 committed:
hw18.1

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/AddItemToBucketController.java:
```diff
@@ -26,10 +26,11 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        Long userId = (Long) req.getSession(true).getAttribute("userId");
```
поле Long USER_ID = 1L можна видалити

Mykhaylo12 replied:
OK

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/CompleteOrderController.java:
```diff
@@ -30,6 +30,6 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
-        resp.sendRedirect(req.getContextPath() + "/orders?user_id=" + bucket.getUserId());
+        resp.sendRedirect(req.getContextPath() + "/servlet/orders?user_id=" + bucket.getUserId());
     }
 }
```
додай пусту строку вкінці файлу

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/LoginController.java:
```diff
@@ -0,0 +1,40 @@
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.*;
```
не використовуємо автоімпорт

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/LoginController.java:
```diff
@@ -0,0 +1,40 @@
+    private static UserService userService;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
занадто довга строка
запускай код стайл перед тим як робиш ПР

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/controller/RegistrationController.java:
```diff
@@ -24,10 +22,17 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
-        userService.create(newUser);
-        resp.sendRedirect(req.getContextPath() + "/allUsers");
+        newUser.setName(req.getParameter("name"));
+        User user=userService.create(newUser);
```
```suggestion
        User user = userService.create(newUser);
```

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/dao/UserDao.java:
```diff
@@ -14,4 +15,8 @@
 
     boolean delete(User user);
+
+    User login(String login, String password) throws AuthenticationException;
```
```suggestion
    User login(String login);
```
про це писала в чат групи

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/web/MainServlet.java:
```diff
@@ -0,0 +1,4 @@
+package mate.academy.internetshop.web;
+
+public class MainServlet {
```
нам потрібен цей клас?

Mykhaylo12 replied:
delete

Sofasmile commented on src/main/java/mate/academy/internetshop/web/filters/AuthenticationFilter.java:
```diff
@@ -0,0 +1,57 @@
+import java.io.IOException;
+import java.util.Optional;
+import java.util.logging.Logger;
+import javax.servlet.*;
```
?

Mykhaylo12 replied:
ok

Mykhaylo12 committed:
1)поле Long USER_ID = 1L видалив в AddItemToBucketController.java
2)додав пусту строку вкінці файлу CompleteOrderController.java
3)LoginController.java забрав автоімпорт
4)Строка 17 в LoginController.java була більше 100 сиволів. Виправлено
5)Стосовно дз, давайте ще раз уточнимо. На рівні дао ми робимо метод Optional<User> findByLogin(String login) який просто шукає юзера за логіном і повертає опшинал, а на рівні сервісу уже перевіряємо валідність його пароля і кидаємо ексепшин. Тобто на рівні сервісу має бути метод User login(login, password).
Зроблено
6)Видалив MainServlet.java
7)Забрав автоімпорт в AuthenticationFilter.java

Mykhaylo12 committed:
Implemented tracking to log file. Everything works well.
Already tested. Information for log4j-application file:
2020-01-15 16:31:10 INFO  InjectInitializer:15 - Dependency injection started
2020-01-15 16:36:09 INFO  InjectInitializer:15 - Dependency injection started
2020-01-15 16:36:09 INFO  InjectInitializer:16 - Test

Mykhaylo12 committed:
fixed LoginController class

Sofasmile commented on log4j-application.log:
```diff
@@ -0,0 +1,4 @@
+2020-01-15 16:31:10 INFO  InjectInitializer:15 - Dependency injection started
```
цей файл краще додати в gitignore щоб він не пушився в ПР

і наступного разу на кожне завдання потрібно створювати нову гілку!

Mykhaylo12 replied:
Додав. Тепер мій файл гітіґнор виглядає так:
.idea/*
target/*
*.iml
log4j-application.log

Sofasmile commented on src/main/java/mate/academy/internetshop/lib/Injector.java:
```diff
@@ -8,15 +8,18 @@
             classes.addAll(getClasses(PROJECT_MAIN_PACKAGE));
         } catch (ClassNotFoundException | IOException e) {
-            e.printStackTrace();
+            LOGGER.error("Injector problems", e);
```
краще тут також кинути RuntimeExc

Mykhaylo12 replied:
ok

Sofasmile commented on src/main/java/mate/academy/internetshop/service/dao/BucketServiceImpl.java:
```diff
@@ -69,7 +68,9 @@ public void clear(Bucket bucket) {
     @Override
     public List<Item> getAllItems(Bucket bucket) {
-        return bucketDao.get(bucket.getBucketId()).orElseThrow(NoSuchElementException::new).getItems();
+        return bucketDao.get(bucket.getBucketId())
```
давай видалимо bucketDao

Mykhaylo12 replied:
Вибач ,але я не розумію, чому потрібно видалити bucketDao і що замість нього зробити

Mykhaylo12 replied:
розібрався

Mykhaylo12 committed:
18.2
Реализовать добавление товара в корзину (через HttpSession)- DONE
Реализовать удаление товара из корзини (через HttpSession)- DONE
Реализовать отображение списка заказов только тех, которые пренадлежат залогированному юзеру (убрать user_id, который был в ДЗ 16, использовать httpSession)- DONE
Посмотреть список товаров в корзине (убрать user_id который был в ДЗ 16, использовать httpSession)- DONE

Mykhaylo12 committed:
1)Injector.java -добавив рантайм ексепшин
2)BucketServiceImpl.java-забрав bucketDao line72

Sofasmile commented on src/main/webapp/WEB-INF/views/login.jsp:
```diff
@@ -6,8 +6,10 @@
+<style  type="text/css"><%@include file="/WEB-INF/styles/util.css"%></style>
 <html>
-<head>
+<head>mvn
```
mvn - тут можна видалити

Sofasmile commented on .gitignore:
```diff
@@ -1,3 +1,4 @@
-*.iml
\ No newline at end of file
+*.iml
+log4j-application.log
```
додай пусту строку

Mykhaylo12 replied:
ok

Sofasmile reviewed (approved):
для того щоб файл log4j-application.log пропав з ПР потрібно зробити так як показано тут
https://www.youtube.com/watch?v=b2Qlqva4-H4&list=PL7FuXFaDeEX2JdShqx3URwdvqf6ow8G7H&index=11&t=0s

Mykhaylo12 committed:
Update .gitignore

Mykhaylo12 committed:
delete mvn line12
